app/models/manageiq/providers/inventory/persister/builder/physical_infra_manager.rb

```diff
@@ -48,9 +48,33 @@ def physical_storages
           add_common_default_values
         end
 
-        def canisters
+        def physical_switches
+          add_properties(:manager_ref => %i(uid_ems))
           add_common_default_values
         end
+
+        # TODO! serial_number wasn't in old refresh's find_key
+        # but there can be more canisters for one storage
+        # http://sysmgt.lenovofiles.com/help/index.jsp?topic=%2Fcom.lenovo.lxca_restapis.doc%2Frest_apis_reference.html
+        # In doc, there is UUID, but not used in this provider
+        # # Tmp solution with serial_number works for specs
+        def canisters
+          add_properties(:manager_ref => %i(physical_storage serial_number))
+          add_properties(:manager_ref_allowed_nil => %i(serial_number))
+        end
+
+        # TODO! serial_number wasn't in old refresh's find_key
+        # Like canisters, physical disks are not uniquely identified in old refresh
+        # Tmp solution with serial_number works for specs
+        def physical_disks
+          add_properties(:manager_ref => %i(physical_storage serial_number))
+          add_properties(:manager_ref_allowed_nil => %i(serial_number))
```

Comment:
Same comment as above

Reply:
unfortunatelly this is not the same. Physical disks doesn't have unique identifier, I'll look what they used in recent PR, but I think they used the same column

Reply:
@agrare discussion was there: https://github.com/ManageIQ/manageiq-schema/pull/285 (same as for canisters)